invenio_app_ils/permissions.py

```diff
@@ -89,6 +97,73 @@ def __init__(self, record):
         )
 
 
+class LoanExtendPermission(Permission):
+    """Return Permission to evaluate if the user can extend the loan."""
+
+    def __init__(self, record):
+        """Constructor."""
+        self.record = record
+
+        # NOTE: extra validation if the user is not librarian or admin
+        if current_user.id == int(record["patron_pid"]):
```

Comment:
Can I propose another version more similar to the other permissions? I think it might be more readable, but let me know what you think.

Transform this into a function:
```
def loan_extend_circulation_permission(loan):
    """ explanation """
    if not current_user or not current_user.id:
        abort(401)

    if current_user.id == loan["patron_pid"]:
        validate_automatic_loan_extension(loan)

    return is_loan_owner_permission(loan["patron_pid"])
```
where `is_loan_owner_permission` is the already existing function `circulation_permission` renamed.

This is because I find it a bit weird to have all these checks in an object constructor and also because we can extract and expose the loan automatic extension business logic to easily test it and eventually re-use it.

WDYT?

Reply:
I was trying to avoid having everything in the constructor, will give it a try for sure.